components/shapediver/ui/ViewportAcceptRejectButtons.tsx

```diff
@@ -0,0 +1,136 @@
+	size?: string | number;
+	color?: string;
+}
+
+interface StyleProps {
+	groupProps?: Partial<GroupProps>;
+	buttonProps?: Partial<ButtonProps>;
+	iconProps?: IconProps;
+	textProps?: Partial<TextProps>;
+}
+
+const defaultStyleProps: Partial<StyleProps> = {
+	groupProps: {
+		justify: "center",
+		w: "auto",
+		wrap: "nowrap",
+		p: "xs",
+	},
+	buttonProps: {
+		variant: "light",
+	},
+	iconProps: {},
+	textProps: {
+		size: "md",
+	},
+};
+
+type ViewportAcceptRejectButtonsComponentThemePropsType = Partial<StyleProps>;
+
+export function ViewportAcceptRejectButtonsComponentThemeProps(
+	props: ViewportAcceptRejectButtonsComponentThemePropsType,
+): MantineThemeComponent {
+	return {
+		defaultProps: props,
+	};
+}
+
+interface Props {
+	sessionIds: string[];
+}
+
+function ViewportAcceptRejectButtons(
+	props: Props & ViewportAcceptRejectButtonsComponentThemePropsType,
+) {
+	const {sessionIds, ...styleProps} = props;
+	// style properties
+	const {groupProps, buttonProps, iconProps, textProps} = useProps(
+		"ViewportAcceptRejectButtonsComponent",
+		defaultStyleProps,
+		styleProps,
+	);
+
+	const showButtons = useMemo(() => sessionIds.length > 0, [sessionIds]);
+
+	// Use a more selective selector that only re-renders when relevant changes occur
+	const parameterChanges = useShapeDiverStoreParameters(
+		useCallback(
+			(state) => {
+				if (!showButtons) return [] as IParameterChanges[];
+
+				return Object.keys(state.parameterChanges)
+					.filter((id) => sessionIds.includes(id))
+					.reduce((acc, id) => {
+						acc.push(state.parameterChanges[id]);
+						return acc;
+					}, [] as IParameterChanges[])
+					.sort((a, b) => a.priority - b.priority);
+			},
+			[sessionIds, showButtons],
+		),
+	);
+
+	const hasChanges = useMemo(
+		() => parameterChanges.length > 0,
+		[parameterChanges],
+	);
+	const disableChangeControls = useMemo(
+		() => parameterChanges.some((c) => c.executing),
+		[parameterChanges],
+	);
+
+	const acceptChanges = useCallback(async () => {
+		for (let index = 0; index < parameterChanges.length; index++) {
+			await parameterChanges[index].accept();
+		}
+	}, [parameterChanges]);
+
+	const rejectChanges = useCallback(() => {
+		parameterChanges.forEach((c) => c.reject());
+	}, [parameterChanges]);
+
+	// If there are no parameter changes to be confirmed, don't render anything
+	if (!showButtons || !hasChanges) {
+		return null;
+	}
+
+	return (
+		<Group {...groupProps}>
+			<Button
+				leftSection={<Icon type={IconTypeEnum.Check} {...iconProps} />}
+				onClick={acceptChanges}
+				disabled={disableChangeControls}
+				{...buttonProps}
+			>
+				<Text {...textProps}>Accept</Text>
+			</Button>
+			<Button
+				leftSection={<Icon type={IconTypeEnum.X} {...iconProps} />}
+				onClick={rejectChanges}
+				disabled={disableChangeControls}
+				{...buttonProps}
+			>
+				<Text {...textProps}>Reject</Text>
+			</Button>
+		</Group>
+	);
+}
+
+export default React.memo(ViewportAcceptRejectButtons);
```

Comment:
@LemToUp ahhh, I didn't know about React.memo, we should probably start using it much more.